sunpy/instr/goes.py

```diff
@@ -583,16 +698,533 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+    the temperature and emission measure values correspond, the
+    radiated losses integrated over time are also calculated.
+
+    Parameters
+    ----------
+    temp : `~astropy.units.Quantity`
+        Array containing the temperature of the coronal plasma at
+        different times.  Units=[MK]
+
+    em : `~astropy.units.Quantity`
+        Array containing the emission measure of the coronal plasma
+        at the same times corresponding to the temperatures in temp.
+        Must be same length as temp.  Units=[cm**-3]
+
+    obstime : (optional) array-like of datetime objects
+        Array of measurement times to which temperature and
+        emission measure values correspond.  Must be same length
+        as temp and em.  If this keyword is set, the integrated
+        radiated energy is calculated.
+
+    force_download : (optional) bool
+        If True, the GOES radiative loss data file is downloaded.  It is
+        important to do this if a new version of the files has been
+        generated due to a new CHIANTI version being released or the
+        launch of new GOES satellites.
+        Default=False
+
+    download_dir : (optional) string
+        The directory to download the GOES radiative loss data file to.
+        Default=SunPy default download directory
+
+    Returns
+    -------
+    rad_loss_out : `dict` of `~astropy.units.quantity.Quantity` objects
+        Contains the following keys.
+
+        | "rad_loss_rate" - radiative loss rate of the soft X-ray-emitting
+           plasma across all wavelengths corresponding to temperatures and
+           emission measures in temp and em Quantity inputs.
+        | "rad_loss_cumul" - cumulative radiative losses as a function of
+          time.  (Only if obstime kwarg is NOT None.)
+        | "rad_loss_int" - total radiative losses as a function of time.
+          (Only if obstime kwarg is not None.)  Array containing radiative
+          loss rates of the coronal plasma corresponding to temperatures and
+          emission measures in temp and em arrays.
+
+    Notes
+    -----
+    This function calls a csv file containing a table of radiative loss
+    rate per unit emission measure at various temperatures.  The
+    appropriate values are then found via interpolation.  This table
+    was generated using CHIANTI atomic physics database employing the
+    methods of Cox & Tucker (1969).  Coronal abundances, a default
+    density of 10**10 cm**-3, and ionization equilibrium of
+    Mazzotta et al. (1998) were used.
+
+    References
+    ----------
+    .. [1] Cox, D. P., Tucker, W. H. 1969, ApJ, 157, 1157
+    .. [2] Mazzotta, P., Mazzitelli, G., Colafrancesco, S., & Vittorio, N.
+       1998, A&AS, 133, 339
+
+    Examples
+    --------
+    >>> from astropy.units.quantity import Quantity
+    >>> temp = Quantity([11.0, 11.0], unit="MK")
+    >>> em = Quantity([4.0e+48, 4.0e+48], unit="cm**(-3)")
+    >>> rad_loss = _calc_rad_loss(temp, em)
```

Comment:
I'm not sure... should we tell how to import functions when explained the function itself?